recipe/session/sessionFunctions.go

```diff
@@ -61,53 +54,74 @@ func createNewSessionHelper(recipeImplHandshakeInfo *sessmodels.HandshakeInfo, c
 	if err != nil {
 		return sessmodels.CreateOrRefreshAPIResponse{}, err
 	}
+
 	return resp, nil
 }
 
-func getSessionHelper(recipeImplHandshakeInfo *sessmodels.HandshakeInfo, config sessmodels.TypeNormalisedInput, querier supertokens.Querier, parsedAccessToken ParsedJWTInfo, antiCsrfToken *string, doAntiCsrfCheck, containsCustomHeader bool) (sessmodels.GetSessionResponse, error) {
-	err := getHandshakeInfo(&recipeImplHandshakeInfo, config, querier, false)
-	if err != nil {
-		return sessmodels.GetSessionResponse{}, err
+func getSessionHelper(config sessmodels.TypeNormalisedInput, querier supertokens.Querier, parsedAccessToken sessmodels.ParsedJWTInfo, antiCsrfToken *string, doAntiCsrfCheck, alwaysCheckCore bool) (sessmodels.GetSessionResponse, error) {
+	var accessTokenInfo *AccessTokenInfoStruct = nil
+	var err error = nil
+	combinedJwks, jwksError := sessmodels.GetCombinedJWKS()
+	if jwksError != nil {
+		if !defaultErrors.As(jwksError, &errors.TryRefreshTokenError{}) {
+			return sessmodels.GetSessionResponse{}, jwksError
+		}
 	}
 
-	var accessTokenInfo *accessTokenInfoStruct = nil
-	foundASigningKeyThatIsOlderThanTheAccessToken := false
-	for _, key := range recipeImplHandshakeInfo.GetJwtSigningPublicKeyList() {
+	accessTokenInfo, err = GetInfoFromAccessToken(parsedAccessToken, *combinedJwks, config.AntiCsrf == AntiCSRF_VIA_TOKEN && doAntiCsrfCheck)
+	if err != nil {
+		if !defaultErrors.As(err, &errors.TryRefreshTokenError{}) {
+			return sessmodels.GetSessionResponse{}, err
+		}
+
+		payload := parsedAccessToken.Payload
 
-		accessTokenInfo, err = getInfoFromAccessToken(parsedAccessToken, key.PublicKey, recipeImplHandshakeInfo.AntiCsrf == antiCSRF_VIA_TOKEN && doAntiCsrfCheck)
-		if err != nil {
-			if !defaultErrors.As(err, &errors.TryRefreshTokenError{}) {
-				return sessmodels.GetSessionResponse{}, err
-			}
+		expiryTimeInPayload, expiryOk := payload["expiryTime"]
+		timeCreatedInPayload, timeCreatedOk := payload["timeCreated"]
 
-			payload := parsedAccessToken.Payload
+		if !expiryOk || !timeCreatedOk {
+			return sessmodels.GetSessionResponse{}, err
+		}
```

Comment:
These should also be inside the < v3 branch: `expiryTime` and `timeCreated` are not present on the v3 tokens, meaning this will always fail for v3 tokens. While we do this anyway after branching, it's cleaner that way.